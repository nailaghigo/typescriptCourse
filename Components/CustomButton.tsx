import React from 'react';
import { Text, StyleSheet, Pressable} from 'react-native';

interface Props {
    onPress: any,
    text: string,
    bgColor: string,
    fgColor: string
  }
const CustomButton : React.FC<Props> = ({onPress, text, bgColor, fgColor}) => {
  return (
    <Pressable
      onPress={onPress}
      style={[
        styles.container,
        bgColor ? {backgroundColor: bgColor} : {},
      ]}>
      <Text
        style={[
          styles.text,
          fgColor ? {color: fgColor} : {},
        ]}>
        {text}
      </Text>
    </Pressable>
  );
};

const styles = StyleSheet.create({
  container: {
    width: '95%',
    alignSelf: 'center',
    padding: 15,
    marginVertical: 5,
    alignItems: 'center',
    borderRadius: 5,
  },

  container_PRIMARY: {
    backgroundColor: '#EB9960',
  },

  container_SECONDARY: {
    borderColor: '#3B71F3',
    borderWidth: 2,
  },

  container_TERTIARY: {},

  text: {
    fontWeight: 'bold',
    color: 'white',
  },

  text_SECONDARY: {
    color: '#3B71F3',
  },

  text_TERTIARY: {
    color: 'gray',
  },
});

export default CustomButton;