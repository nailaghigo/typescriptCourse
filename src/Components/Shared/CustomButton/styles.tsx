import {StyleSheet} from 'react-native';

const styles = StyleSheet.create({
  container: {
    width: '95%',
    alignSelf: 'center',
    padding: 15,
    marginVertical: 5,
    alignItems: 'center',
    borderRadius: 5,
  },

  container_PRIMARY: {
    backgroundColor: '#EB9960',
  },

  text: {
    fontWeight: 'bold',
    color: 'white',
  },
});

export default styles;
