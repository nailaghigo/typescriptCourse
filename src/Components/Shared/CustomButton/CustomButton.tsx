import React from 'react';
import {Text, Pressable} from 'react-native';
import styles from './styles';

interface Props {
  onPress: any;
  text: string;
  bgColor: string;
  fgColor: string;
  testID?: string;
}
const CustomButton: React.FC<Props> = ({
  onPress,
  text,
  bgColor,
  fgColor,
  testID,
}) => {
  return (
    <Pressable
      onPress={onPress}
      testID={testID}
      style={[styles.container, bgColor ? {backgroundColor: bgColor} : {}]}>
      <Text style={[styles.text, fgColor ? {color: fgColor} : {}]}>{text}</Text>
    </Pressable>
  );
};

export default CustomButton;
