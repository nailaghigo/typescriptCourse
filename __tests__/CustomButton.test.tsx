import 'react-native';
import React from 'react';
import CustomButton from '../src/Components/Shared/CustomButton/CustomButton';
import {render, waitFor} from '@testing-library/react-native';

describe('CustomButton', () => {
  it('renders correctly', async () => {
    const {getByText, queryByTestId, toJSON} = render(
      <CustomButton
        onPress={() => {}}
        text="Test text"
        testID="example-button"
        bgColor={''}
        fgColor={''}
      />,
    );

    await waitFor(() => expect(queryByTestId('example-button')).toBeTruthy());

    expect(getByText('Test text')).toBeTruthy();
    expect(toJSON()).toMatchSnapshot();
  });
});
