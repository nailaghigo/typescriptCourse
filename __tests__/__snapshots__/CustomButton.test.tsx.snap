// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomButton renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "alignSelf": "center",
        "borderRadius": 5,
        "marginVertical": 5,
        "padding": 15,
        "width": "95%",
      },
      Object {},
    ]
  }
  testID="example-button"
>
  <Text
    style={
      Array [
        Object {
          "color": "white",
          "fontWeight": "bold",
        },
        Object {},
      ]
    }
  >
    Test text
  </Text>
</View>
`;
